### ISSUE TITLE
Kitty doesn't always refresh the display when switching tabs

### ISSUE THREAD

**Issue by dhanak** [kovidgoyal/kitty#7954]
**Describe the bug**

I have multiple tabs in my kitty window. When I switch tabs, the display is not always refreshed immediately, only when I do something else (e.g., start typing). The title of the terminal changes, though, so I can tell that the tab switch was executed.

**To Reproduce**

This issue is not reproducible reliably, it just occurs every once in a while. I haven't been able to observe any specific circumstance that must exist in order for the issue to manifest. It does, however, occur fairly often, several times a day.

**Screenshot**

I will try to make a screenshot of this next time it happens.

**Environment details**
```
kitty 0.36.3 (44039baa39) created by Kovid Goyal
Linux ci0156 6.1.0-26-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.112-1 (2024-09-30) x86_64
Debian GNU/Linux 12 ci0156 /dev/tty

Running under: X11
OpenGL: '4.6 (Core Profile) Mesa 22.3.6' Detected version: 4.6
Frozen: True
Fonts:
  medium: MesloLGMDZNF-Regular: /home/david/.fonts/MesloLGMDZNerdFont-Regular.ttf:0
          Features: ()
    bold: MesloLGMDZNF-Bold: /home/david/.fonts/MesloLGMDZNerdFont-Bold.ttf:0
          Features: ()
  italic: MesloLGMDZNF-Italic: /home/david/.fonts/MesloLGMDZNerdFont-Italic.ttf:0
          Features: ()
      bi: MesloLGMDZNF-BoldItalic: /home/david/.fonts/MesloLGMDZNerdFont-BoldItalic.ttf:0
          Features: ()
Paths:
  kitty: /home/david/.local/kitty.app/bin/kitty
  base dir: /home/david/.local/kitty.app/lib/kitty
  extensions dir: /home/david/.local/kitty.app/lib/kitty-extensions
  system shell: /bin/zsh
Loaded config files:
  /home/david/.config/kitty/kitty.conf

Config options different from defaults:
active_tab_font_style     (True, False)
cursor_blink_interval     (0.0, kitty.options.utils.EasingFunction(), kitty.options.utils.EasingFunction())
enabled_layouts           ['splits', 'stack']
font_family               MesloLGMDZ Nerd Font
font_size                 11.5
shell_integration         frozenset({'no-cursor'})
tab_bar_edge              1
tab_bar_style             powerline
tab_powerline_style       slanted
tab_title_template        {fmt.fg.red}{bell_symbol}{activity_symbol}{fmt.fg.tab}{title}{fmt.fg.orange}{' ' if layout_name == 'stack' else ''}
text_composition_strategy legacy
Removed mouse actions:
	left click ungrabbed →  mouse_handle_click selection link prompt
Added shortcuts:
	ctrl+alt++ →  change_font_size all +1.0
	ctrl+alt+- →  change_font_size all -1.0
	ctrl+alt+0 →  goto_tab 10
	ctrl+alt+1 →  goto_tab 1
	ctrl+alt+2 →  goto_tab 2
	ctrl+alt+3 →  goto_tab 3
	ctrl+alt+4 →  goto_tab 4
	ctrl+alt+5 →  goto_tab 5
	ctrl+alt+6 →  goto_tab 6
	ctrl+alt+7 →  goto_tab 7
	ctrl+alt+8 →  goto_tab 8
	ctrl+alt+9 →  goto_tab 9
	ctrl+alt+= →  change_font_size all +1.0
	ctrl+alt+backspace →  change_font_size all 0
	ctrl+alt+enter →  launch --location=split
	ctrl+alt+z →  toggle_layout stack
	ctrl+shift+alt+g →  launch --stdin-source=@last_cmd_output --stdin-add-formatting --type=window less -R
	kitty_mod+p > l →  kitten hints --type hash --program "launch --type=overlay --env=LESS= --env=DELTA_PAGER='less -LR~' git show", kitten hints --type path --program "launch --type=overlay bat --paging=always", kitten hints --type url --program @, kitten hints --customize-processing redmine-hints.py, kitten hints --multiple --type path --program -, kitten hints --type path --program "launch --type=background emacsclient --no-wait", kitten hints --type linenum --linenum-action=background emacsclient --no-wait +{line} {path}, kitten hints --type hash --add-trailing-space always --program -, kitten hints --type path, kitten hints --type path --add-trailing-space always --program -, kitten hints --type hyperlink, kitten hints --type word --program -, kitten hints --type line --program -
	shift+down →  move_window down
	shift+f5 →  launch --location=hsplit --cwd=last_reported
	shift+f6 →  launch --location=vsplit --cwd=last_reported
	shift+f7 →  layout_action rotate
	shift+left →  move_window left
	shift+right →  move_window right
	shift+up →  move_window up
Removed shortcuts:
	kitty_mod+p > f →  kitten hints --type hyperlink, kitten hints --type linenum, kitten hints --type hash --program -, kitten hints --type word --program -, kitten hints --type line --program -, kitten hints --type path, kitten hints --type path --program -
Changed shortcuts:
	kitty_mod++ →  no-op
	kitty_mod+- →  no-op
	kitty_mod+= →  no-op
	kitty_mod+backspace →  no-op
	kitty_mod+enter →  launch --location=split --cwd=last_reported
	kitty_mod+n →  launch --type=os-window
	kitty_mod+t →  launch --type=tab --location=after
Colors:
	active_border_color       #75715e   
	active_tab_background     #75715e   
	active_tab_foreground     #272822   
	background                #272822   
	color0                    #272822   
	color1                    #f92672   
	color10                   #a6e22e   
	color11                   #e6db74   
	color12                   #66d9ef   
	color13                   #fd971f   
	color14                   #ae81ff   
	color15                   #f8f8f2   
	color2                    #a6e22e   
	color3                    #e6db74   
	color4                    #66d9ef   
	color5                    #fd971f   
	color6                    #ae81ff   
	color7                    #f8f8f2   
	color8                    #75715e   
	color9                    #f92672   
	cursor                    #f8f8f2   
	foreground                #f8f8f2   
	inactive_border_color     #75715e   
	inactive_tab_background   #31332b   
	inactive_tab_foreground   #75715e   
	selection_background      #f8f8f2   
	selection_foreground      #272822   
	tab_bar_background        #31332b   
	url_color                 #f8f8f2   

Important environment variables seen by the kitty process:
	PATH                                /home/david/.cargo/bin:/home/david/.local/bin:/home/david/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
	LANG                                en_US.UTF-8
	EDITOR                              emacsclient
	SHELL                               /bin/zsh
	DISPLAY                             :0
	USER                                david
	LC_MEASUREMENT                      en_GB.UTF-8
	LC_MONETARY                         en_GB.UTF-8
	LC_NUMERIC                          en_GB.UTF-8
	LC_PAPER                            en_GB.UTF-8
	LC_TIME                             en_GB.UTF-8
	XDG_DATA_DIRS                       /usr/share/gnome:/home/david/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share/
	XDG_RUNTIME_DIR                     /run/user/1000
	XDG_CURRENT_DESKTOP                 GNOME
	XDG_MENU_PREFIX                     gnome-
	XDG_SESSION_CLASS                   user
	XDG_SESSION_DESKTOP                 gnome-xorg
	XDG_SESSION_TYPE                    x11
```

**Additional context**

I noticed this behavior quite a while ago, around version 0.30 or so, I can't recall exactly. I have tried to reproduce it with `--config NONE`, without success, but since reproduction is uncertain anyhow, I can't be sure if the issue is caused by some setting or not.

**Comment by kovidgoyal** [comment]
If you can find a reproducer I will be happy to take a look, as it
stands there isn't anything actionable here. Feel free to update this
issue if you can find a reproducer or any more relevant information. I
can tell you that I use kitty tabs personally all the time in X11 and
have never noticed this.

**Comment by dhanak** [comment]
It occurred several times this morning. I noticed that it is enough to wait 2-3 seconds, and then the refresh/redraw occurs. I still don't know why it happens: I do the tab switch ten-twenty times in a row, and then suddenly, there it is. But after a few seconds, it redraws, and the next tab switch is okay again. Odd.

**Comment by dhanak** [comment]
I just noticed that this phenomenon is not restricted to switching tabs: when running app is redrawing the entire screen, kitty doesn't always refresh the screen immediately. I noticed this while running tmux on a remote server (I know your opinion on tmux, but I'm using it for session management and persistence), and zoom in/out of panes.

Odd, really. The title bar refreshes, but the contents of the kitty window only follow after a few seconds.
